conn/doc.go

```diff
@@ -4,6 +4,45 @@
 
 // Package conn defines core interfaces for protocols and connections.
 //
-// Subpackages implements all the interfaces that can be used to connects
-// peripherals.
+// Subpackages implements all the supported interfaces that can be used to
+// connects peripherals.
+//
+// Concepts
+//
+// periph uses 3 layered concepts for interfacing:
+//
+//     Bus → Port → Conn
+//
+// Not every subpackage expose all 3 concepts. In fact, most packages don't.
+// For example, SPI doesn't exxpose Bus as the OSes generally only expose the
+// Port, that is, a CS line must be selected right upfront to get an handle.
+// For I²C, there's no Port to configure, so selecting a "slave" address is
+// sufficient to jump directly from a Bus to a Conn.
+//
+// Bus
+//
+// A Bus is a multi-point communication channel where one "master" and multiple
+// "slaves" communicate together. In the case of periph, the Bus handle is
+// assumed to be the "master". The "master" generally initiates communications
+// and selects the "slave" to talk to.
+//
+// As the "master", selecting a "slave" over a Bus converts the Bus into a Port.
```

Comment:
suggestion: `As the "master" selects a "slave" over a bus, a virtual port is automatically created` or something like that. The purpose of my suggestion is to highlight the fact that the developer doesn't have anything to do and that the bus and the port are available (AFAIK), in other words, the bus doesn't convert into a port and disappears. I'm putting myself in the choose of someone who just bought a rpi for instance and is discovering the various connections.